web/src/components/metrics/metrics-content.tsx

```diff
@@ -58,93 +59,62 @@ export const MetricsContent: React.FC<{
   const { t } = useTranslation('plugin__netobserv-plugin');
 
   const metricTitle = React.useCallback(() => {
-    if (metricFunction === 'rate') {
-      return t('Flows rate');
-    } else if (metricType) {
-      switch (metricFunction) {
-        case 'avg':
-          return t('Average {{type}} (1m frame)', { type: metricType });
-        case 'max':
-          return t('Max {{type}} (1m frame)', { type: metricType });
-        case 'sum':
-          return t('Total {{type}}', { type: metricType });
-        default:
-          return '';
-      }
-    } else {
-      console.error('metricType cannot be undefined');
-      return '';
+    switch (metricFunction) {
+      case 'last':
+        return t('Latest {{type}} rate', { type: metricType });
+      case 'avg':
+        return t('Average {{type}} rate', { type: metricType });
+      case 'max':
+        return t('Max {{type}} rate', { type: metricType });
+      case 'sum':
+        return t('Total {{type}}', { type: metricType });
+      default:
+        return '';
     }
   }, [metricFunction, metricType, t]);
 
   const chart = React.useCallback(() => {
-    function truncate(input: string) {
-      const length = doubleWidth ? 64 : showDonut ? 10 : 18;
-      if (input.length > length) {
-        return input.substring(0, length / 2) + '…' + input.substring(input.length - length / 2);
-      }
-      return input;
-    }
-
-    function truncateParts(input: string) {
-      if (input.includes('.')) {
-        const splitted = input.split('.');
-        const result: string[] = [];
-        splitted.forEach(s => {
-          result.push(truncate(s));
-        });
-        return result.join('.');
-      }
-      return truncate(input);
-    }
-
-    function getName(m: TopologyMetric) {
-      switch (scope) {
-        case MetricScopeOptions.HOST:
-          const srcNode = truncateParts(getMetricName(m, scope, true, t));
-          const dstNode = truncateParts(getMetricName(m, scope, false, t));
-
-          return data?.host
-            ? m.SrcK8S_HostName === data.host
-              ? `${t('To')} ${dstNode}`
-              : `${t('From')} ${srcNode}`
-            : `${srcNode} -> ${dstNode}`;
-        case MetricScopeOptions.NAMESPACE:
-          const srcNamespace = truncateParts(getMetricName(m, scope, true, t));
-          const dstNamespace = truncateParts(getMetricName(m, scope, false, t));
+    // function truncate(input: string) {
```

Comment:
@jpinsonneau I commented this out because I find it was too much truncated (which makes it sometimes hard to understand what a metric is), and in most cases it's displayed correctly without truncation.
The other issue with truncation is that there's no tooltip showing the whole text.
But I guess in some cases not truncating really screws up the display ? .. so I don't know, maybe bring it back but with bigger limit value?

Reply:
Yes PF / Victory is not really responsive on that part. I can add a `Tooltip` on top of `ChartLabel` to show the full text on hover + play with length

Reply:
restored: 5c8bbfb4413713dfb9c4c86cb6babc7f479dd364